### PR TITLE
Disallow null pixels data for CompressedTex{Sub}Image{2D|3D}

### PR DESCRIPTION
**Pull request by qkmiao**
Related chromium fix is at: https://codereview.chromium.org/1527723003/.

### CODE REVIEW

File: sdk/tests/deqp/functional/gles3/es3fNegativeTextureApiTests.js [hunk 1]
```diff
@@ -488,7 +490,7 @@ goog.scope(function() {
 
 
             bufferedLogToConsole('gl.INVALID_VALUE is generated if imageSize is not consistent with the format, dimensions, and contents of the specified compressed image data.');
-            gl.compressedTexImage2D(gl.TEXTURE_2D, 0, gl.COMPRESSED_RGBA8_ETC2_EAC, 0, 0, 0, null);
+            gl.compressedTexImage2D(gl.TEXTURE_2D, 0, gl.COMPRESSED_RGBA8_ETC2_EAC, 0, 0, 0, new Uint8Array(1));
```
Comment (kenrussell) [inline sdk/tests/deqp/functional/gles3/es3fNegativeTextureApiTests.js]: Why is this one different than the others?

File: sdk/tests/deqp/functional/gles3/es3fNegativeTextureApiTests.js [hunk 2]
```diff
@@ -2043,7 +2045,7 @@ goog.scope(function() {
             this.expectError (gl.NO_ERROR);
 
             bufferedLogToConsole('gl.INVALID_VALUE is generated if imageSize is not consistent with the format, dimensions, and contents of the specified compressed image data.');
-            gl.compressedTexSubImage2D(gl.TEXTURE_2D, 0, 0, 0, 0, 0, gl.COMPRESSED_RGBA8_ETC2_EAC, null);
+            gl.compressedTexSubImage2D(gl.TEXTURE_2D, 0, 0, 0, 0, 0, gl.COMPRESSED_RGBA8_ETC2_EAC, new Uint8Array(1));
```
Comment (kenrussell) [inline sdk/tests/deqp/functional/gles3/es3fNegativeTextureApiTests.js]: Same question here.

File: sdk/tests/deqp/functional/gles3/es3fNegativeTextureApiTests.js [hunk 3]
```diff
@@ -2641,7 +2643,7 @@ goog.scope(function() {
             gl.bindTexture (gl.TEXTURE_2D_ARRAY, texture);
 
             bufferedLogToConsole('gl.INVALID_VALUE is generated if imageSize is not consistent with the format, dimensions, and contents of the specified compressed image data.');
-            gl.compressedTexImage3D(gl.TEXTURE_2D_ARRAY, 0, gl.COMPRESSED_RGBA8_ETC2_EAC, 0, 0, 0, 0, null);
+            gl.compressedTexImage3D(gl.TEXTURE_2D_ARRAY, 0, gl.COMPRESSED_RGBA8_ETC2_EAC, 0, 0, 0, 0, new Uint8Array(1));
```
Comment (kenrussell) [inline sdk/tests/deqp/functional/gles3/es3fNegativeTextureApiTests.js]: And here.

**Comment by kenrussell** [comment]
Looks fine overall. Couple of minor questions.

File: sdk/tests/deqp/functional/gles3/es3fNegativeTextureApiTests.js (hunk 1 above)
Comment (zhenyao) [inline sdk/tests/deqp/functional/gles3/es3fNegativeTextureApiTests.js]: The original deqp test uses -1 here, which is impossible for WebGL.

Uint8Array(1) is a reasonable replacement of -1.

Comment (qkmiao) [inline sdk/tests/deqp/functional/gles3/es3fNegativeTextureApiTests.js]: In order to generate INVALID_VALUE for zero-size texture but pixels data size isn't zero.

**Comment by kenrussell** [comment]
OK. This seems reasonable. CC'ing @jasomobica . Merging now.

**Pull request merged**